### PR TITLE
Add Solana program address configuration

### PR DESCRIPTION
**Pull request by Sabnock01**
## Summary
- Add a Solana deployment helper for deriving CPMM and initializer config addresses from a supplied program address set.
- Allow initialize launch and CPMM migration account builders to derive accounts from supplied CPMM/migrator program IDs while preserving existing devnet defaults.
- Add Solana tests covering default and supplied program address behavior.

## Testing
- `pnpm format:check`
- `pnpm typecheck`
- `pnpm lint`
- `pnpm test:solana`

### CODE REVIEW

File: test/solana/deployment.test.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        allowBuy: true,
+        allowSell: true,
+        hookProgram: CUSTOM_PROGRAMS.cpmmHookProgram,
+        hookFlags: 0,
+        hookPayload: new Uint8Array(),
+        migratorInitPayload: new Uint8Array(),
+        migratorMigratePayload: new Uint8Array(),
+        hookRemainingAccountsHash: initializer.EMPTY_REMAINING_ACCOUNTS_HASH,
+        migratorInitRemainingAccountsHash:
+          initializer.EMPTY_REMAINING_ACCOUNTS_HASH,
```
Comment (z80dev) [inline test/solana/deployment.test.ts]: Minor question: is it okay for this custom CPMM path to run/pass with the empty init hash here?

The builder appends the CPMM migrator init remaining accounts (`[cpmmMigrationState, cpmmConfig]`) later, and the IDL docs say `migrator_init_remaining_accounts_hash` commits to those remaining accounts. So I would have expected this test to use `computeRemainingAccountsHash([cpmmMigrationState, cpmmConfig])` rather than `EMPTY_REMAINING_ACCOUNTS_HASH`, or at least assert that an empty hash is intentionally accepted here.

Maybe I’m missing a runtime detail, but this looks like it could let the test pass while the actual initialize-launch instruction would have a non-empty account tail with an empty committed init hash.
